firestore/integration_test_internal/src/validation_test.cc

```diff
@@ -472,6 +472,77 @@ TEST_F(ValidationTest,
   EXPECT_NE(Firestore::GetInstance("foo"), nullptr);
 }
 
+TEST_F(ValidationTest,
+       FirestoreGetInstanceCalledMultipleTimeReturnSameInstance) {
+  {
+    Firestore* instance1 = Firestore::GetInstance();
+    Firestore* instance2 = Firestore::GetInstance();
+    EXPECT_EQ(instance1, instance2);
+  }
+  {
+    Firestore* instance1 = Firestore::GetInstance(app());
+    Firestore* instance2 = Firestore::GetInstance(app());
+    EXPECT_EQ(instance1, instance2);
+  }
+  {
+    Firestore* instance1 = Firestore::GetInstance("foo");
+    Firestore* instance2 = Firestore::GetInstance("foo");
+    EXPECT_EQ(instance1, instance2);
+  }
+  {
+    Firestore* instance1 = Firestore::GetInstance(app(), "foo");
+    Firestore* instance2 = Firestore::GetInstance(app(), "foo");
+    EXPECT_EQ(instance1, instance2);
+  }
+}
+
+TEST_F(ValidationTest,
+       differentFirestoreGetInstanceCanGetSameDefaultFirestoreInstance) {
+  Firestore* instance1 = Firestore::GetInstance();
+  Firestore* instance2 = Firestore::GetInstance(app());
+  Firestore* instance3 = Firestore::GetInstance("(default)");
+  Firestore* instance4 = Firestore::GetInstance(app(), "(default)");
+
+  EXPECT_EQ(instance1, instance2);
+  EXPECT_EQ(instance1, instance3);
+  EXPECT_EQ(instance1, instance4);
+  EXPECT_EQ(instance2, instance3);
```

Comment:
There is a transitive relationship in equals. If 1==2 and 1==3, then we can infer 2==3. I don't think you need addition `EXPECT_EQ`.

Reply:
True. Checked all permutations for precaution. Will remove this.